lib/serializer/cbor_serializer.ex

```diff
@@ -81,4 +88,23 @@ defmodule Cashu.Serializer.V4 do
 
   defp maybe_handle_tag(%CBOR.Tag{} = data), do: data.value
   defp maybe_handle_tag(data), do: data
+
+  defp to_v4_token(map) do
+    %Token{
+      mint: map["m"],
+      unit: map["u"],
+      memo: map["d"],
+      token: parse_tokens(map["t"])
+    }
+  end
+
+  defp parse_tokens(tokens) do
+    Enum.flat_map(tokens, &parse_proofs(&1))
+  end
+
+  defp parse_proofs(proof) do
+    proof["p"]
+    |> Enum.map(&Map.put(&1, "i", proof["i"]))
```

Comment:
you could pass `p` and `i` to `parse_proofs` without fetching them each time, or pattern match them at line 105, or even pass them as args in line 102. Not a big difference but might be a little more efficient.

Reply:
Thank you for your suggestions. Let me know if what I came up is readable.